src/txmempool.cpp

```diff
@@ -632,7 +632,7 @@ void CTxMemPool::check(const CCoinsViewCache *pcoins) const
     uint64_t innerUsage = 0;
 
     CCoinsViewCache mempoolDuplicate(const_cast<CCoinsViewCache*>(pcoins));
-    const int64_t spendheight = GetSpendHeight(mempoolDuplicate);
+    const int64_t spendheight = WITH_LOCK(::cs_main, return g_chainman.m_blockman.GetSpendHeight(mempoolDuplicate));
```

Comment:
The reason why I thought it'd be okay to lock `cs_main` after `mempool.cs` was because that's basically what's happening in GetSpendHeight prior to this changeset. As in:

1. We lock `mempool.cs` at the start of `CTxMemPool::check`
2. We call `GetSpendHeight`, which locks `cs_main`

Are you saying that this was wrong all along?

Reply:
Resolved by: https://github.com/bitcoin/bitcoin/pull/20972